### ISSUE TITLE
Auto-magically & retroactively have posts that are only ±1, &c, converted to reactions & deleted

### ISSUE THREAD

**Issue by TPS** [isaacs/github#640]
To fix issues like #18 not loading because of death by 1300 (__❗❕❗__) :+1:s (& then having to be refiled in #604 to work-around), GitHub should now automatically & retroactively find all posts everywhere that consist ___solely___ of +1 (:+1:, 👍, +100, :100:, +∞, :shipit:{?}, [mega-👍s](https://camo.githubusercontent.com/240a9da71ed71c6dce5897bb7a78adce6f3ff206/68747470733a2f2f6d6564696130646b2d612e616b616d616968642e6e65742f33362f33352f36363033646335613932393231303462343463333439623835623561616637612d352d6372617a792d66616e2d7468656f726965732d746861742d6d616b652d746f74616c2d73656e73652e6a7067), & related should count, too), -1 (& related), &/or all other supplied reactions, then should convert them to reactions on the top/previous non-reaction-only post (repo owner's choice) & then delete from view or thread outright.  It'd instantly solve such subscription-spamming (does _anyone_ __sane__ need [e-mail notification on these posts](https://github.com/isaacs/github/issues/631#issuecomment-205008905)?!), as well.

To give credit where due, @eddiemonge [proposed this way back in April 23, 2015](https://github.com/isaacs/github/issues/377#issuecomment-95697073) & @andreynering [on March 30, 2016](https://github.com/dear-github/dear-github/issues/152#issuecomment-203701405)!

---
[↓ Don't just react: Contact GitHub supporting this, ___also___, & report back! ↓](https://github.com/contact)

**Comment by TPS** [comment]
:+1::heart:

**Comment by TPS** [comment]
If this is ever implemented, the intermediate post should evaporate & convert to reactions on original post. :stuck_out_tongue_winking_eye:

**Comment by TPS** [comment]
Response from GitHub:
>Thanks for reaching out. Yes it might be nice to go back and clean up all the +1 comments etc. and convert them into reactions.

>I can't make any promises, but I have definitely added it to the feature request list and will share it with the team for consideration.

>Regards,
Daniel
@danayel
GitHub Support

**Comment by TPS** [comment]
@sergei-ivanov I'm honored by your support! :bow:

**Comment by TPS** [comment]
~~Does anyone know if #584 is dead because of this, too?  Or is it something else?~~  Nevermind, but why is it private?

**Comment by TPS** [comment]
Another issue that would benefit enormously from this is #215! 😵

**Comment by The-Compiler** [comment]
I think there are two different issues:

- converting *existing* +1 into reactions (what this issue is about)
- preventing *new* +1 - either by introducing some UI change to nudge users to use reactions instead, or by not permitting posts containing exactly `+1` at all.

**Comment by TPS** [comment]
@The-Compiler I _suppose_ 1 could split this into 2, but neither part is particularly satisfactory on its own, & I think the same backend implementation could handle both more easily than just 1-or-the-other.

**Comment by TPS** [comment]
[From](https://github.com/isaacs/github/issues/283#issuecomment-213295431) @radek-holy:
>@TPS, as you can see, there are people who intentionally use comments instead of reactions. They have their reasons. So I believe that if GitHub starts to convert these comments into reactions, they will start write such comments that would not fit the GitHub's regexp since their goal is to post a comment (and to express support), not just to express support. So, my question was more about those emails that are intended to work around this issue regardless of their content…

I'll reply here as not to spam #283: Practically,  such non-content support 👍s are spam on the channel for everyone _else_ who's subscribed (so inherently selfish), & causes other problems as detailed above. *I*'d rather have those folks put the effort into a _useful_ response/comment, `xor` just minimally react, not the hybrid spam. Besides, all this convesion is to be @ the repo _admins'_ discretion, as mentioned above.

**Comment by Kubo2** [comment]
Response from GitHub:

> Thanks for getting in touch! It isn't possible to convert emoji comments to reactions at this time. We've added your idea and rationale to our internal Feature Request List so the team can see it. It might be something we support in the future, but we can't promise when it will happen.

> All the best,
GitHub Staff

**Comment by TPS** [comment]
Hey, @Mottie, if you want a really good challenge, [userscript](https://github.com/Mottie/GitHub-userscripts) _this_! :bow:

**Comment by Mottie** [comment]
<img title=":_ cry :" src="https://cloud.githubusercontent.com/assets/136959/13512303/638c0a7e-e15f-11e5-83ab-bb898da7d97d.gif"> I don't think it's possible.

From looking at the reactions API, you can [create a reaction](https://developer.github.com/v3/reactions/#create-reaction-for-an-issue), but it will only be *your* reaction; there is no way that I can see to add more than one thumbs up to the top issue. Also, you would need to have authorization to [edit](https://developer.github.com/v3/issues/comments/#edit-a-comment) or [delete](https://developer.github.com/v3/issues/comments/#delete-a-comment) "+1" comments from the thread.

I really wish there was some way to add a webhook or integration where when someone only posts a "+1" or equivalent, it automatically adds it as a reaction to the OP in their name and deletes the comment. It wouldn't solve this issue of converting retroactive "+1s", but it may stop people from their current bad habits.

**Comment by TPS** [comment]
I didn't mean as an actual implementation of the whole functionality, but merely what 1 sees:  Something like (on top post) "862 👍s removed & should be added here" or some such, but less tacky wording. 😜

**Comment by Mottie** [comment]
The [GitHub issue comments](https://github.com/Mottie/GitHub-userscripts/wiki/GitHub-issue-comments) userscript will hide "+1" comments - it also hides stuff like "+1!!!", "++1", "+9999", "+1 please", ":+1:", ":100:", etc. It won't hide the comment if there is more text than that.

I just updated the userscript to include a count of how many comments were hidden. So, it won't count comments that have a "+1" with an explanation...

In [#215](https://github.com/isaacs/github/issues/215#issuecomment-110343438), the userscript initially hides 112 comments, but after you click "view more" - and check & uncheck the "Hide +1" option (I still need to fix this) - it shows that a total of 197 comments are hidden.

![2016-05-22 11_00_43-insert automatically generated table of contents toc on rendered markdown files](https://cloud.githubusercontent.com/assets/136959/15455107/da04fd8e-200f-11e6-9039-2e06c5c359c2.png)

This is probably the best I can offer at this time <img title=":_bipolar:" src="https://cloud.githubusercontent.com/assets/136959/13512241/15d8362c-e15f-11e5-8c32-35b78deb9017.gif">.

**Comment by TPS** [comment]
@Mottie The only addition I can think of that's _possible_, perhaps: add, e.g., those 197 👍s, to the top-post's 👍-count _or_ as an additional "& scraped: 197 :+1:" box in that line.  Still, you :metal:!

**Comment by TPS** [comment]
Oops!

**Comment by Mottie** [comment]
Ok, latest update will now add a little extra comment to the top post

![with reactions](https://cloud.githubusercontent.com/assets/136959/15488579/57ee9846-211d-11e6-9919-e41fea3bd38d.png)

If there isn't a +1 reaction, it adds a comment to the post

![no reactions](https://cloud.githubusercontent.com/assets/136959/15488623/d5105080-211d-11e6-9774-4970283b5f6b.png)

I ended up tweaking the code a little so the hidden +1 count may be a little different. Also, it automatically updates after more comments are loaded using the "View more" button.

**Comment by TPS** [comment]
@Mottie You, sir, deserve my deepest respect. :bow:

**Comment by Mottie** [comment]
Oh weird... I guess the thumbs up button can have different values. That's why the "256 (from hidden comments)" isn't inside the button. I'll fix that when I get some time today.

The `256` is because when you hit the "1560 more items not show View more" button, it only adds 200 more comments... scroll down and you'll find a "1360 more items not shown View more" button...

**Comment by Mottie** [comment]
Ok, the userscript has been updated again... it now makes sure that repeat +1-ers only get one vote.

**Comment by marcelmfs** [comment]
+1

**Comment by TPS** [comment]
C'mon, @marcelmfs, you can do better than <kbd>[___`that❕`___](https://github.com/contact)

**Comment by marcelmfs** [comment]
I thought that the +1 post would be automatically converted to :+1: reaction, as per the title of the issue.

**Comment by TPS** [comment]
@marcelmfs Only if you yourself had already convinced @GitHub to implement it.… 😜  & [my official test case is second post, above](#issuecomment-207202008).

**Comment by arshaw** [comment]
I've created an alternative UI for browsing and ranking Github Issues. See my newly created project:

[ghi-scoreboard](https://github.com/arshaw/ghi-scoreboard) (Github project)
[example scoreboard](http://fullcalendar.io/issues/) (fullcalendar)
[blog announcement](http://fullcalendar.io/blog/2016/07/renovated-issue-tracker/)

One of its features is the ability to use these lame +1 comments for ranking purposes (albeit aggregated on the server-side, periodically). If you set `aggregateComments:true`, can you sort by `plusComments` (+1 or 👍  within a comment) or `plusScore`, which combines comments with real reactions.

Hope it comes in handy for some of you.

**Comment by TPS** [comment]
Seriously _nice_ work, @arshaw!  I really appreciate the work you put into it, & truly hope @GitHub takes notice!

**Comment by asbjornu** [comment]
Can't the subscribe button just count as a vote?

**Comment by TPS** [comment]
@asbjornu Not really.…  E.g., *I*'m actually subscribed to more issues than I've 👍ed, even some I've 👎ed, because I want to be notified on updates, whether or not I've any reaction (some because I'm indecisive, but others are interesting w/o my having any reaction), or because I don't have enough standing in the project such that my reaction is relevant to the devs.

**Comment by asbjornu** [comment]
@TPS, good point. But my experience from other issue trackers tells me that being able to up and downvote issues is something that are integral to the business domain of issue tracking and that it would make a lot of sense for GitHub to implement it. The possibility to react with emoji to comments is a step in the right direction, but does not give everything up and downvoting gives in other systems such as Trello, Jira, YouTrack, UserVoice, etc.

**Comment by TPS** [comment]
@Mottie  [GitHub issue comments userscript](https://github.com/Mottie/GitHub-userscripts/wiki/GitHub-issue-comments) seems not to work at all for me anymore, but I'm not sure if because it's from my side, or something @GitHub.  Is it still functional for you? :cry:

**Comment by Mottie** [comment]
It's working for me in this issue... hiding 2 +1s in the popup but only showing +1 from hidden comments in the first.

**Comment by TPS** [comment]
It must be my setup, then.  Thanks, @Mottie.

**Comment by Mithgol** [comment]
Remember that a “+1” comment is not only a reaction, but also a (somewhat greater) alternative to the “Subscribe” button. The following alternatives are currently available:

* hit a “Subscribe” button to receive notifications in a longer https://github.com/notifications list,

* write a “+1” comment to receive notifications in a shorter https://github.com/notifications/participating list.

**Comment by TPS** [comment]
@Mithgol [As mentioned, earlier](https://github.com/isaacs/github/issues/640#issuecomment-213444485), all those reasons are particularly selfish. If 1 wants to _actually_ participate, let'em do so!  If 1 doesn't, don't!  But don't pervert the tools for 1's own ends to the detriment of others!

**Comment by TPS** [comment]
@Mottie I see ":+1: 1 (from hidden comments)" on _this_ issue, though https://github.com/isaacs/github/issues/640#issuecomment-207202008 & https://github.com/isaacs/github/issues/640#issuecomment-225580472 should be 2, right?  Both comments are properly disappeared, though.

**Comment by Mottie** [comment]
Internally, the code keeps a list of the names of all commenters, so duplicate posts with a +1 are ignored.

**Comment by TPS** [comment]
Yes, but the 2 :+1: posts are from myself & @marcelmfs.  Are you saying @marcelmfs both posted & :+1:ed the original post?

**Comment by Mottie** [comment]
Your original post (the first one) has a +1, thumbs up, 100... and all that stuff that counts as a +1 by the algorithm. So the second post is considered a duplicate.

**Comment by TPS** [comment]
But, then what happened to @marcelmfs's :+1:?

**Comment by dessant** [comment]
I've made a GitHub App that deletes +1 comments, and encourages people to use reactions instead. :tada: 

https://probot.github.io/apps/reaction/

**Comment by TPS** [comment]
@dessant Fantastic! Now only if @GitHub / @Microsoft would install this ___globally___!

A suggestion/clarification:  Does that "Edit Issues" permission allow to accomplish the rest of this RFE, adding the reaction meant by the now-deleted post?

**Comment by dessant** [comment]
@TPS, the [permission](https://developer.github.com/v3/apps/permissions/#permission-on-issues) is used to edit/delete comments. GitHub Apps cannot submit reactions, that is reserved to OAuth Apps, though those need to be installed individually by the user.

There is no clear way to solve this ourselves at the org level, short of creating a fake reaction bar at the bottom of the issue body that imitates the real one and increments reaction counts there. That would still be a muddy solution, because you cannot guess the intentions of the user, did they react to a recent comment with a +1, or are they agreeing with the goals of the entire issue? You would see otherwise misguided issues that received a reasonable comment along the way appear to have the support of the community.

A sensible approach would be to educate users, show a warning about the consequences of a +1 comment when they are about to submit the post, and suggest the use of reactions instead. I agree this could be implemented globally by GitHub.

**Comment by lukehefson** [comment]
Hello all! I'm a [PM at GitHub](https://blog.github.com/2018-08-28-announcing-paper-cuts/) and I just wanted to leave a note here to say that we shipped something that should hopefully help with this: https://blog.github.com/changelog/2018-09-24-hiding-repeat-comments/

I know that this solution doesn't absolutely address the title above – so I don't expect it to close this issue out – but I hope that it goes some way to help 🧡.

I'd love to know your feedback. How can we iterate to make this better? For the slightly different problem of stopping people from posting "just +1" posts in the first place – we're currently investigating what we can do there too!

**Comment by TPS** [comment]
@lukehefson Thanks _very_ much for your attention! :bow:  This is _indeed_ a very welcome stopgap toward implementing my (😜) issue, & it works beautifully. A couple RFC/Es:
- Could it work across duplicates that _aren't_ consecutive?
- Does/could it work to suppress the _e-mail/notifications_ of such comments?

**Comment by marcelmfs** [comment]
+1 👍

**Comment by TPS** [comment]
@JayFoxRox, @KenOrb, @ArzTe: I noticed that y'all expressed some concerns ([via reactions on OP, above](https://github.com/isaacs/github/issues/640#issue-146807201)), but haven't otherwise detailed via comments.  I'd like to explicitly invite y'all to do so, especially since we've @GitHub's 👀 via @lukehefson!

@marcelmfs C'mon, ___again___?  [Haven't we been through this already⁉️ 😝](https://github.com/isaacs/github/issues/640#issuecomment-225580472)

**Comment by kenorb** [comment]
Some new feature has been implemented, by grouping multiple "+1" posts, but I can't find the article about it.

**Comment by kenorb** [comment]
+1

**Comment by kenorb** [comment]
+1

**Comment by agos** [comment]
@TPS thanks to this thread I can confirm that it does not suppress email notifications 😞
